include/SFML/System/Vector3.inl

```diff
@@ -56,6 +56,75 @@ z(static_cast<T>(vector.z))
 }
 
 
+////////////////////////////////////////////////////////////
+template <typename T>
+T Vector3<T>::length() const
+{
+    static_assert(std::is_floating_point_v<T>, "Vector3::length() is only supported for floating point types");
+
+    return std::hypot(x, y, z);
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+constexpr T Vector3<T>::lengthSq() const
+{
+    return dot(*this);
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+Vector3<T> Vector3<T>::normalized() const
+{
+    static_assert(std::is_floating_point_v<T>, "Vector3::normalized() is only supported for floating point types");
+
+    assert(*this != Vector3<T>());
+    return (*this) / length();
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+constexpr T Vector3<T>::dot(const Vector3<T>& rhs) const
+{
+    return x * rhs.x + y * rhs.y + z * rhs.z;
+}
+
+
+////////////////////////////////////////////////////////////
+template <typename T>
+constexpr Vector3<T> Vector3<T>::cross(const Vector3<T>& rhs) const
+{
+    Vector3<T> out;
+    out.x = (y * rhs.z) - (z * rhs.y);
+    out.y = (z * rhs.x) - (x * rhs.z);
+    out.z = (x * rhs.y) - (y * rhs.x);
+
+    return out;
```

Comment:
Minor nitpick, maybe directly return a `Vector3` instead of constructing a zero vector first.
```cpp
return Vector3<T>(
    (y * rhs.z) - (z * rhs.y),
    (z * rhs.x) - (x * rhs.z),
    (x * rhs.y) - (y * rhs.x)
);
```

Reply:
Yeah, definitely a cleaner way to do it. I'm happy to go add this change👍